parallel-install/pkg/postuninstaller/postuninstaller_test.go

```diff
@@ -0,0 +1,430 @@
+package postuninstaller
+
+import (
+	"context"
+	"fmt"
+	"github.com/avast/retry-go"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/config"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/logger"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/preinstaller"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/preinstaller/mocks"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/fake"
+	"testing"
+)
+
+func TestPostUninstaller_UninstallCRDs(t *testing.T) {
+
+	scheme := runtime.NewScheme()
+	cfg := getTestingConfig()
+	retryOptions := getTestingRetryOptions()
+
+	t.Run("should not uninstall CRDs", func(t *testing.T) {
+		t.Run("when no resources of any kind are present on a cluster", func(t *testing.T) {
+			// given
+			client := fake.NewSimpleDynamicClientWithCustomListKinds(scheme, fixCrdGvrMap())
+			resourceManager := &mocks.ResourceManager{}
+			uninstaller := getPostUninstaller(cfg, resourceManager, client, retryOptions)
+
+			// when
+			output, err := uninstaller.UninstallCRDs()
+
+			// then
+			requireNoErrorAndNoOutput(t, err, output)
+			resourceManager.AssertNumberOfCalls(t, "DeleteResource", 0)
+		})
+
+		t.Run("when CRDs not labeled by Kyma are present on a cluster", func(t *testing.T) {
+			// given
+			crd1 := fixCrdResourceWith("crd1", "apiextensions.k8s.io", "v1beta1", "origin", "unknown")
+			crd2 := fixCrdResourceWith("crd2", "apiextensions.k8s.io", "v1beta1", "origin", "unknown")
+			crd3 := fixCrdResourceWith("crd3", "apiextensions.k8s.io", "v1beta1", "origin", "unknown")
+			client := fake.NewSimpleDynamicClientWithCustomListKinds(scheme, fixCrdGvrMap())
+			applyMockObj(t, client, crd1, fixCrdGvrV1Beta1())
+			applyMockObj(t, client, crd2, fixCrdGvrV1Beta1())
+			applyMockObj(t, client, crd3, fixCrdGvrV1Beta1())
+			resourceManager := &mocks.ResourceManager{}
+			uninstaller := getPostUninstaller(cfg, resourceManager, client, retryOptions)
+
```

Comment:
the above makes this tests too large imho, could we refactor/extract this?

Reply:
I will put creation & application into common function wherever applicable (where all resources get the same properties). It should make it slightly shorter then!